### PR TITLE
Adding HOLE

### PR DESCRIPTION
**Pull request by ianmkenney**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by ianmkenney** [comment]
@IAlibay Mind taking a look? macOS build seems fine, but Linux is failing since `quadmath` isn't found. `libgfortan` for macOS is at v5 while Linux is at v3. Installing the `gfortran` compiler from `conda-forge` locally reproduces the errors.

**Comment by IAlibay** [comment]
I _think_ it might be picking up the wrong gfortran path, could you try running `which gfortran` and `gfortran -v` in your script to see what the outputs are?

File: recipes/hole2/build.sh
Comment (IAlibay) [inline recipes/hole2/build.sh:4]:
```suggestion
make FC=${FC} CC=${CC}
```
Try this?

**Comment by IAlibay** [comment]
I am willing to act as maintainer for this package.

**Comment by ianmkenney** [comment]
@orbeckst do you have any idea why the macOS built hangs?

**Comment by orbeckst** [comment]
no idea, sorry, maybe @IAlibay ??

**Comment by IAlibay** [comment]
> do you have any idea why the macOS built hangs?

By the looks of things the macOS build is hanging because qplot is expecting in a user input? (i.e. https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=648616&view=logs&j=e35d4f76-8ff2-5536-d795-df91e63eb9f7&t=fa7b4b17-b6ff-5c9c-8cfc-15f888c92310&l=2073).

**Comment by orbeckst** [comment]
@ianmkenney there's old docs for `labqpt` at http://www.csb.yale.edu/userguides/graphics/hole/doc/hole_d07.html : maybe just feed a PDB and some of the other inputs and just check that it groks them?

We can bundle the gramicidin A PDB.

**Comment by ianmkenney** [comment]
@IAlibay @orbeckst it looks like the macOS build doesn't even let me interact. Comparing its output to the Linux build, it does nothing with the `echo -e "\n\n\n" | labqpt`. It also seems that it doesn't allow you to use the program like `labqpt input.pdb output.qpt`.

**Comment by ianmkenney** [comment]
@orbeckst could you compile hole2 from our fork on your Mac, `make PREFIX=${WHEREVER} install-all` and let me know how the output for `./labqpt` compares to the build on Azure?  Additionally, are you able to use either `echo -e "EXIT" | ./labqpt` or `./labqpt <&-` to get a `0` return code as you can in Linux?

**Comment by orbeckst** [comment]
Output on macOS:
```
$ ./labqpt
 This program allows the user to produce a QUANTA
  binary 3D plot file containing labels for selected atoms
   read from a pdb format co-ordinate file.
 Copyright 1993,1997 by Oliver Smart and Birkbeck College
 Copyright 2004 by Oliver Smart
 Program modification number 2.2 001

 For help on HOLE suite see  http://www.holeprogram.org/

 HOLE release 2.2.005 (07 August 2016)

 usage subject to Apache License, Version 2.0
 see http://www.apache.org/licenses/LICENSE-2.0

 Program qplot includes an extension to the standard qpt format.'   You can specify an integer to indicate where each string
   should be placed on the page in a qplot output file
       1 indicates below to the left
       2 ............... centred etc.
       key to remember  789
                        456 (like a numeric keypad)
                        123

 Please enter input pdb co-ordinate (old) filename
 defaults <input.pdb> ext:<.pdb> (abort by EXIT or ^D) :
```
(Note that '''input.pdb''' and '''.pdb''' are inverted.)
* I can abort with `^D` and `EXIT<RETURN>`and get `STOP FORTRAN STOP - labqpt normal completion.` when entering from the terminal.
* `echo -e` does not work on macOS. WIth `printf "EXIT\n" | ./labqpt` it waits for stdin and the ignores the stdin input (on;y `^C` gets me out).
*  `./labqpt <&-` same: ignores stdin

So far I have not found a way to get it to read from a pipe.

**Comment by IAlibay** [comment]
Is the check that `labqpt` works a hard requirement at the moment? I.e. can we live with just checking that the binary exists and get this merged, then work on the issue after deployment? (just thinking about timelines and how long it can take for packages to filter through on cf during busy periods).

**Comment by orbeckst** [comment]
Not a hard requirement. This was one of the optional tools anyway. 

It works in manual mode when I literally type it in but it does not like input from a pipe. I am staring at the code for the virtual terminal handling... but I should not waste more time.

In short: I agree with @IAlibay : just check that the binary exists and raise an issue for later.

**Comment by ianmkenney** [comment]
@orbeckst @IAlibay sounds good, I agree

File: recipes/hole2/meta.yaml
Comment (orbeckst) [inline recipes/hole2/meta.yaml:2]: any chance we can set this back to 2.3.0 to match the planned 2.3.0 release https://github.com/osmart/hole2/issues/5 ?

File: recipes/hole2/meta.yaml
Comment (orbeckst) [inline recipes/hole2/meta.yaml:11]: Will this be changed to https://github.com/osmart/hole2/ as soon as we make a release https://github.com/osmart/hole2/issues/5 ?

**Review by orbeckst — CHANGES_REQUESTED** [review]
I think we don't want the feedstock to be merged until we have a 2.3.0 release of hole2 https://github.com/osmart/hole2/issues/5 , right?

So once all test of the feedstock pass, we can merge PR https://github.com/osmart/hole2/pull/2 , then make the release, and then merge the feedstock.

File: recipes/hole2/meta.yaml
Comment (ianmkenney) [reply recipes/hole2/meta.yaml:2]: yeah that's the plan for right now

File: recipes/hole2/meta.yaml
Comment (ianmkenney) [reply recipes/hole2/meta.yaml:11]: Yes

**Comment by orbeckst** [comment]
I am willing to act as a maintainer of the package.

**Comment by orbeckst** [comment]
@ianmkenney you can check more bullets: license and maintainers and everything related to static libs.

**Comment by orbeckst** [comment]
@ianmkenney where are we installing the radius files, i.e. the `rad` directory? (`make install-data`)?

**Review by orbeckst — APPROVED** [review]
all good!

**Comment by IAlibay** [comment]
https://github.com/orgs/conda-forge/teams/help-python this should be good to review :)

**Comment by orbeckst** [comment]
@conda-forge/help-python this should be good to review :)

Thank you!

**Comment by orbeckst** [comment]
I am sorry, I wrongly asked for a review from the Python team (out of habit...) but this is *not* a python package.

I am not sure how to remove the Python tag .

**Comment by BastianZim** [comment]
Now you should be able to ping the team you need. :)

> I am not sure how to remove the Python tag .

I think only team members can.

**Comment by orbeckst** [comment]
@conda-forge/staged-recipes , please review this PR. Thank you!

**Comment by BastianZim** [comment]
That chocked the bot

**Comment by orbeckst** [comment]
Thank you for helping @BastianZim . Apologies for creating extra work.

**Comment by IAlibay** [comment]
@conda-forge/staged-recipes Sorry to bother - would it be possible to get a review / merge here? All the maintainers have given their approval and CI seems to be going fine.

**Comment by IAlibay** [comment]
@conda-forge/help-c-cpp Would someone from this team maybe might be willing to review this? It's a mostly Fortran package (with a couple of C files) with a make build

File: recipes/hole2/meta.yaml
Comment (jaimergp) [inline recipes/hole2/meta.yaml:3]:
```suggestion
{% set build_number = 0 %}
```

The build number should never be connected to the source versioning. It's used to track changes in the recipe for the _same_ source, so this defeats the purpose. Can you please revert to just `0`?

File: recipes/hole2/meta.yaml
Comment (jaimergp) [inline recipes/hole2/meta.yaml:51]:
```suggestion
  home: https://www.holeprogram.org
```

File: recipes/hole2/build.sh
Comment (jaimergp) [inline recipes/hole2/build.sh:6]: I think we need to set more variables here so we are using the ones set by the conda-forge compilers infrastructure.

[All of these](https://github.com/osmart/hole2/blob/b8fdc748df2da19867a0b21778fd1115a118dbb2/src/Makefile#L20-L32) should (at least) include the ones the [activation scripts are setting at build time](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=653001&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=846):

```shell
# excerpt
...
+FFLAGS=-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/hole2-2.3.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix
...
```

Comment (ianmkenney) [reply recipes/hole2/build.sh:6]: If I'm understanding correctly, for lines 5 and 6, all we need to do is add `FFLAGS=${FFLAGS}` and `CFLAGS=${CFLAGS}`. Is that what you're referring to?

Comment (orbeckst) [reply recipes/hole2/build.sh:6]: Can you just use appending `+=` instead of assignment `:=`
```make
FFLAGS += -fd-lines-as-comments -fbackslash -std=legacy -static-libgfortran
CFLAGS += -static-libgfortran
```
to ensure that our options are added to whatever the environment sets (given that [make turns env vars into makevars](https://www.gnu.org/software/make/manual/html_node/Environment.html)?

File: recipes/hole2/meta.yaml
Comment (ianmkenney) [reply recipes/hole2/meta.yaml]: fixed in latest commit

File: recipes/hole2/meta.yaml
Comment (ianmkenney) [reply recipes/hole2/meta.yaml]: fixed in latest commit

File: recipes/hole2/LICENSE
Comment (ianmkenney) [reply recipes/hole2/LICENSE]: removed in latest commit

File: recipes/hole2/build.sh
Comment (orbeckst) [reply recipes/hole2/build.sh:6]: My other question is what to do with program names such as `ar`. The current makefile has https://github.com/osmart/hole2/blob/b8fdc748df2da19867a0b21778fd1115a118dbb2/src/Makefile#L28 
```
AR := ar -rcv
```
and the [build env sets](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=653001&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=852)
```
AR=$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-ar
```
I think we need to **rewrite the HOLE Makefile** to *not include the `-rcv` in the command name* and instead put it after invocation like https://github.com/osmart/hole2/blob/b8fdc748df2da19867a0b21778fd1115a118dbb2/src/Makefile#L163-L169
```
$(LIB_NAME): $(FILES) 
	$(AR) -rcv $(LIB_NAME) $(FILES)  
	$(RANLIB) $(LIB_NAME)

.f.o:
	$(FC) $(FFLAGS) -c $*.f
	$(AR) -rcv $(LIB_NAME) $*.o
```
I am not 100% sure if we then just pass the build system's env vars explicitly to the Makefile as
```bash
make ... AR=${AR} ...
```
or if make will automatically take it from the env and use it instead of `AR := ar`.

Comment (orbeckst) [reply recipes/hole2/build.sh:6]: Final comment: in the Makefile we have a few invocations of `$(FC)` such as https://github.com/osmart/hole2/blob/b8fdc748df2da19867a0b21778fd1115a118dbb2/src/Makefile#L195-L196
```makefile
makver.exe: makver.f
	$(FC) makver.f -o makver.exe
```
which currently ignore FFLAGS and friends. Given that the build system insists on setting them, we also need to respect them so to be safe, the above should become
```makefile
makver.exe: makver.f
	$(FC) $(FFLAGS) $(LFLAGS) makver.f -o makver.exe
```
(Need to be checked if it works with the full set of FFLAGS...)

**Comment by ianmkenney** [comment]
@jaimergp ready for review!

**Comment by orbeckst** [comment]
@jaimergp would you mind having another look at the PR, please? We'd really like to move forward with creating the hole2 conda-forge package. Thank you!

**Comment by ianmkenney** [comment]
@jaimergp when you have a moment, could you please have another look at the PR? Thanks!

File: recipes/hole2/build.sh
Comment (isuruf) [inline recipes/hole2/build.sh:7]: CC_FOR_BUILD?

Comment (ianmkenney) [reply recipes/hole2/build.sh:7]: Is this not the compiler we should be using? It's been a while, but it seems at point I believed that the [`GCC` variable was not set](https://github.com/ianmkenney/staged-recipes/commit/204f457f9370887572bbedb4b90c4c683aea0763)

Comment (isuruf) [reply recipes/hole2/build.sh:7]: Use `CC` which is the compiler for the platform we are compiling for. `CC_FOR_BUILD` is the compiler for the platform we are building in.

Comment (ianmkenney) [reply recipes/hole2/build.sh]: Switched it over in the next commit

**Comment by ianmkenney** [comment]
@isuruf seems to compile fine with that change!

**Review by jaimergp — APPROVED** [review]

**Comment by orbeckst** [comment]
Thank you for taking the time @jaimergp , much appreciated!

**Comment by jaimergp** [comment]
I'll merge in 24h if there are no further comments!

**Comment by orbeckst** [comment]
Thank you very much, everyone, and especially @jaimergp !

**Pull request merged**
